tests/test_crypto/test_ledger_apis.py

```diff
@@ -48,56 +54,102 @@ def test_initialisation(self):
+        assert balance == 0, "This must be 0 since the address is wrong"
+        balance = ledger_apis.token_balance(ETHEREUM, fet_address)
+        assert balance == 0, "This must be 0 since the address is wrong"
+
+        with pytest.raises(AssertionError):
+            balance = ledger_apis.token_balance("UNKNOWN", fet_address)
+            assert balance == 0, "Unknown identifier so it will return 0"
+
+    def test_transfer(self):
+        """Test the transfer function for the supported tokens."""
+        private_key_path = os.path.join(CUR_PATH, "data", "eth_private_key.txt")
+        eth_obj = EthereumCrypto(private_key_path=private_key_path)
+        private_key_path = os.path.join(CUR_PATH, 'data', "fet_private_key.txt")
+        fet_obj = FetchAICrypto(private_key_path=private_key_path)
+        ledger_apis = LedgerApis({ETHEREUM: DEFAULT_ETHEREUM_CONFIG,
+                                  FETCHAI: DEFAULT_FETCHAI_CONFIG})
+
+        with mock.patch.object(ledger_apis.apis.get(FETCHAI).tokens, 'transfer',
+                               return_value="97fcacaaf94b62318c4e4bbf53fd2608c15062f17a6d1bffee0ba7af9b710e35"):
+            tx_digest = ledger_apis.transfer(FETCHAI, fet_obj, fet_address, amount=10, tx_fee=10)
+            assert tx_digest is not None
+
+        with mock.patch.object(ledger_apis.apis.get(ETHEREUM).eth, 'getTransactionCount', return_value=5):
+            transaction = {
+                'nonce': ledger_apis.apis.get(ETHEREUM).eth.getTransactionCount(),
+                'chainId': 3,
+                'to': eth_address,
+                'value': 50,
+                'gas': 2000000,
+                'gasPrice': ledger_apis.apis[ETHEREUM].toWei(GAS_PRICE, GAS_ID)
+            }
+            signed = ledger_apis.apis[ETHEREUM].eth.account.signTransaction(transaction, eth_obj._account.privateKey)
+            with mock.patch.object(ledger_apis.apis.get(ETHEREUM).eth.account, 'signTransaction',
+                                   return_value=signed):
+                with pytest.raises(ValueError):
+                    ledger_apis.apis[ETHEREUM].eth.sendRawTransaction(signed.rawTransaction)
+                result = HexBytes('0xf85f808082c35094d898d5e829717c72e7438bad593076686d7d164a80801ba005c2e99ecee98a12fbf28ab9577423f42e9e88f2291b3acc8228de743884c874a077d6bc77a47ad41ec85c96aac2ad27f05a039c4787fca8a1e5ee2d8c7ec1bb6a')
+                with mock.patch.object(ledger_apis.apis.get(ETHEREUM).eth, 'sendRawTransaction',
+                                       return_value=result):
+                    with mock.patch.object(ledger_apis.apis.get(ETHEREUM).eth, "getTransactionReceipt",
+                                           return_value=b'0xa13f2f926233bc4638a20deeb8aaa7e8d6a96e487392fa55823f925220f6efed'):
+                        tx_digest = ledger_apis.transfer(ETHEREUM, eth_obj, eth_address, amount=10, tx_fee=200000)
+                        assert tx_digest is not None
+
+    def test_is_tx_settled(self):
```

Comment:
the same as above, would be better to split this test in multiple tests. At least the ones regarding `fetchai` and the others regarding `ethereum`.